component/prometheus/operator/common/crdmanager.go

```diff
@@ -266,6 +275,20 @@ func (c *crdManager) runInformers(restConfig *rest.Config, ctx context.Context)
 		if ls != labels.Nothing() {
 			opts.DefaultLabelSelector = ls
 		}
+
+		// TODO: Remove custom opts.Mapper when sigs.k8s.io/controller-runtime >= 0.17.0 as `NewDynamicRESTMapper` is the default in that version
+		var err error
+		opts.HTTPClient, err = rest.HTTPClientFor(restConfig)
+		if err != nil {
+			return err
+		}
+
+		opts.Mapper, err = apiutil.NewDynamicRESTMapper(restConfig, opts.HTTPClient)
```

Comment:
Change in mapper is required as the dynamic rest mapper lazy loads mappings. Hence, we can retry calls to `GetInformer` below rather than just failing when calling `cache.New` here.